src/Service/InvitePartnerService.php

```diff
@@ -131,6 +131,71 @@ private function createEmailAndSmsData(ObjectEntity $requester, ObjectEntity $pe
     }//end createEmailAndSmsData()
 
 
+    /**
+     * This function gets the partner details from the given bsn.
+     *
+     * @param ObjectEntity $huwelijkObject The huwelijk from the request.
+     * @param array        $huwelijk       The body from the request.
+     * @param string       $bsn            The bsn of the given partner via email.
+     *
+     * @return array The partner datails as array.
+     */
+    private function invitePartnerLogin(ObjectEntity $huwelijkObject, array $huwelijk, string $bsn): array
+    {
+        $brpSchema      = $this->gatewayResourceService->getSchema('https://vng.brp.nl/schemas/brp.ingeschrevenPersoon.schema.json', 'common-gateway/huwelijksplanner-bundle');
+        $partnerDetails = [];
+
+        $brpPersons = $this->cacheService->searchObjects(null, ['burgerservicenummer' => $bsn], [$brpSchema->getId()->toString()])['results'];
+        if (count($brpPersons) === 1) {
+            $partnerDetails['brpPerson'] = $this->entityManager->find('App:ObjectEntity', $brpPersons[0]['_self']['id']);
+        }//end if
+
+        foreach ($huwelijkObject->getValue('partners') as $huwelijkPartner) {
+            if ($bsn === $huwelijkPartner->getValue('contact')->getValue('subjectIdentificatie')->getValue('inpBsn')) {
+                $partnerDetails['personAssent'] = $huwelijkPartner;
+                $partnerDetails['person']       = $this->assentService->createPerson($huwelijk, $partnerDetails['brpPerson'], $huwelijkPartner->getValue('contact'));
+            }//end if
+
+            if ($bsn !== $huwelijkPartner->getValue('contact')->getValue('subjectIdentificatie')->getValue('inpBsn')) {
+                $partnerDetails['partner'] = $huwelijkPartner->getValue('contact');
+            }//end if
+        }//end foreach
+
+        return $partnerDetails;
+
+    }//end invitePartnerLogin()
+
+
+    /**
+     * This function gets the partner details from the given bsn.
+     *
+     * @param ObjectEntity $huwelijkObject The huwelijk from the request.
+     * @param string       $bsn            The bsn of the given partner via email.
+     *
+     * @return array The partner datails as array.
+     */
+    private function invitePartnerInvite(ObjectEntity $huwelijkObject): array
+    {
+        $partnerDetails = [];
+
+        foreach ($huwelijkObject->getValue('partners') as $huwelijkPartner) {
+            if ($huwelijkPartner->getValue('contact')->getValue('subjectIdentificatie') === false) {
+                $partnerDetails['personAssent'] = $huwelijkPartner;
+                $partnerDetails['person']       = $huwelijkPartner->getValue('contact');
+            }//end if
```

Comment:
(//end if or //end foreach is only necessary if the if statement is long, more than 5 lines long or something like that)